src/marine/profile/profile2ioda.py

```diff
@@ -0,0 +1,260 @@
+        idata['ob_sal_err'] = []
+        idata['ob_tmp'] = []
+        idata['ob_tmp_err'] = []
+
+        for n in range(self.odata['n_obs']):
+
+            lat = self.odata['ob_lat'][n]
+            lon = self.odata['ob_lon'][n]
+            dtg = datetime.strptime(self.odata['ob_dtg'][n], '%Y%m%d%H%M')
+
+            # Diff. between ob. time and time in the file (hrs)!
+            ddtg = (dtg - self.date).seconds/3600.
+
+            for l, lvl in enumerate(self.odata['ob_lvl'][n]):
+
+                idata['ob_lat'].append(lat)
+                idata['ob_lon'].append(lon)
+                idata['ob_dtg'].append(ddtg)
+                idata['ob_lvl'].append(lvl)
+                idata['ob_sal'].append(self.odata['ob_sal'][n][l])
+                idata['ob_sal_err'].append(self.odata['ob_sal_err'][n][l])
+                idata['ob_tmp'].append(self.odata['ob_tmp'][n][l])
+                idata['ob_tmp_err'].append(self.odata['ob_tmp_err'][n][l])
+
+        idata['nlocs'] = len(idata['ob_dtg'])
+
+        self.idata = idata
+
+        return
+
+    def to_ioda(self):
+
+        if self.odata['n_obs'] <= 0:
+            print('No profile observations for IODA!')
+            return
+
+        self._to_nlocs()
+
+        ncid = netcdf.netcdf_file(self.filename + '.nc', mode='w')
+
+        ncid.createDimension('nlocs', self.idata['nlocs']) # total no. of obs
+        ncid.createDimension('nrecs', self.odata['n_obs']) # no. of profiles
+        ncid.createDimension('nvars', 2) # no. of variables
+
+        longitudes = ncid.createVariable('longitude', 'f4', ('nlocs',))
+        longitudes.units = 'degrees_east'
+
+        latitudes = ncid.createVariable('latitude', 'f4', ('nlocs',))
+        latitudes.units = 'degrees_north'
+
+        times = ncid.createVariable('time', 'f4', ('nlocs',))
+        times.long_name = 'observation time from date_time'
+        times.units = 'hours'
+        times.description = 'why does this have to be with a reference to, why not absolute?'
+
+        depths = ncid.createVariable('ocean_depth', 'f4', ('nlocs',))
+        depths.units = 'meters'
+        depths.positive = 'down'
+
+        sals = ncid.createVariable('ocean_salinity@ObsValue', 'f4', ('nlocs',))
+        sals.units = 'g/kg'
+        sals.description = ''
+
+        sal_errs = ncid.createVariable(
+            'ocean_salinity@ObsError', 'f4', ('nlocs',))
+        sal_errs.units = '(g/kg)^2'
+        sal_errs.description = 'Standard deviation of observation error'
+
+        tmps = ncid.createVariable(
+            'insitu_temperature@ObsValue', 'f4', ('nlocs',))
+        tmps.units = 'degree_celcius'
+        tmps.description = ''
+
+        tmp_errs = ncid.createVariable(
+            'insitu_temperature@ObsError', 'f4', ('nlocs',))
+        tmp_errs.units = '(degree_celcius)^2'
+        tmp_errs.description = 'Standard deviation of observation error'
+
+        ncid.date_time = int(datetime.strftime(self.date, '%Y%m%d%H%M'))
```

Comment:
Something is fishy with the global attribute (getting date_time = -59312912). We need yyyymmddhh (no mn) as an integer, how about 
ncid.date_time = int(datetime.strftime(self.date, '%Y%m%d%H%M')[0:10]) <-- That does the job.

Reply:
@guillaumevernieres When one does `int(YYYYMMDDHHMM)`, it is treated as `int32` and you get the fishy value.
If you infact do `np.int64(YYYYMMDDHHMM)`, you will get the integer`YYYYMMDDHHMM`.

So, when you truncate the string YYYYMMDDHHMM to `[0:10]`, you can represent that integer as int32 and thus the seemingly correct value is returned.

This raises the question, what about observations that are going to contain higher frequency of sampling (minutes and seconds)?

There are 2 solutions for getting rid of fishy value; the 1st one you pointed out. The second is: `ncid.date_time = int(datetime.strftime(self.date, '%Y%m%d%H'))`

Reply:
addressed @guillaumevernieres changes.